packages/runtime-core/src/helpers/renderList.ts

```diff
@@ -67,20 +61,12 @@ export function renderList(
   const sourceIsArray = isArray(source)
 
   if (sourceIsArray || isString(source)) {
-    const sourceIsReactiveArray = sourceIsArray && isReactive(source)
-    let needsWrap = false
-    if (sourceIsReactiveArray) {
-      needsWrap = !isShallow(source)
-      source = shallowReadArray(source)
-    }
     ret = new Array(source.length)
-    for (let i = 0, l = source.length; i < l; i++) {
-      ret[i] = renderItem(
-        needsWrap ? toReactive(source[i]) : source[i],
```

Comment:
- The `needsWrap` check here is an optimization of array item access. It should not be removed.
- In addition, performance optimization requires providing benchmarks

Reply:
<img width="602" alt="image" src="https://github.com/user-attachments/assets/4ccb7c3e-7ebe-4b5c-ab8f-4bfca13e2437">
it will be wrap in iterator. so i remove it

Reply:
I know this, so the benchmark is required. We need to make sure that this change actually improves performance.

Reply:
<img width="960" alt="image" src="https://github.com/user-attachments/assets/a68b0682-306c-4bac-9cd3-f9329074be5a">
We can see that the performance improvement of using for of compared to the normal for loop is huge.

Reply:
I think the following is the correct test case
```ts
// renderList current
{
    const rawArray: any[] = []
    for (let i = 0, n = amount; i < n; i++) {
      rawArray.push({ value: i })
    }

    let arr = reactive(rawArray)
    const sourceIsArray = isArray(arr)
    const sourceIsReactiveArray = sourceIsArray && isReactive(arr)
    let needsWrap = false
    if (sourceIsReactiveArray) {
      needsWrap = !isShallow(arr)
      arr = shallowReadArray(arr)
    }

    bench(`normal for loop with shallowReadArray, ${amount} elements`, () => {
      let sum = 0
      effect(() => {
        for (let i = 0; i < arr.length; i++) {
          sum += (needsWrap?toReactive(arr[i]):arr[i]).value
        }
      })
    })
  }

  // renderList with this PR
  {
    const rawArray: any[] = []
    for (let i = 0, n = amount; i < n; i++) {
      rawArray.push({ value: i })
    }
    const arr = reactive(rawArray)

    bench(`for...of loop, ${amount} elements`, () => {
      let sum = 0
      effect(() => {
        for (const item of arr) {
          sum += item
        }
      })
    })
  }
```